src/components/Tooltip/BaseGenericTooltip/index.native.tsx

```diff
@@ -0,0 +1,132 @@
+import React, {useEffect, useMemo, useRef, useState} from 'react';
```

Comment:
Although there's a great progress, this file is still duplicated too much (`BaseGenericTooltip/index.tsx` vs `BaseGenericTooltip/index.native.tsx`). I'll ask some clarifying questions, which might help us figure out how we can abstract things out.